.github/workflows/nightly_smoke_tests.yml

```
@@ -4,25 +4,99 @@ on:
   schedule:
     - cron: "0 0 * * *"
   workflow_dispatch:
+    inputs:
+      sha:
+        description: 'Commit SHA to test'
+        required: false
+        default: ''
+        type: string
 
 jobs:
   smoke_tests:
+    if: github.repository == 'linode/terraform-provider-linode' || github.event_name == 'workflow_dispatch'
     runs-on: ubuntu-latest
 
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
         with:
-          ref: dev
+          fetch-depth: 0
+          submodules: 'recursive'
+          ref: ${{ github.event.inputs.sha || github.ref }}
 
-      - name: Set up go
+      - name: Set up Go
         uses: actions/setup-go@v5
         with:
-          go-version: 'stable'
+          go-version: '1.x'
 
-      - run: go version
+      - name: Install Dependencies
+        run: |
+          make deps
 
       - name: Run smoke tests
-        run: make smoke-test
+        id: smoke_tests
+        run: |
+          make smoke-test
         env:
           LINODE_TOKEN: ${{ secrets.DX_LINODE_TOKEN }}
+
+      - name: Notify Slack
+        if: always()
+        uses: slackapi/slack-github-action@v1.26.0
```

```suggestion
        uses: slackapi/slack-github-action@v1.27.0
```
The 1.27 version of the action is available now